cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -314,7 +315,7 @@ func (rm *AzureResourceManager) appendDeploymentResourcesRecursive(
 			// Operations w/o target data can't be resolved. Ignoring them
 			continue
 		}
-		if *operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment) {
+		if stringutil.PtrValueEquals(operation.Properties.TargetResource.ResourceType, string(AzureResourceTypeDeployment)) {
```

Comment:
We have the pattern of doing the check as:
```
               if operation.Properties.TargetResource == nil {
			// Operations w/o target data can't be resolved. Ignoring them
			continue
		}
```
My personal preference is to continue using that pattern b/c it explicitly check conditions and give us the change to break-fast and even add logs within those breaks/returns about it.

This other pattern where we only do stuff when the conditions are met and ignore all other cases leads to longer conditions.

This is not the first time we are affected by these indeterministic `nil` values from service responses (see this other fix: https://github.com/Azure/azure-dev/pull/1559).  

My preference would be to keep checking for `nil` directly before accessing a `*type` and without auxiliar functions.  I know is longer code, but easier to follow/digest.

Reply:
This won't work here because `TargetResource` is not `nil` but `ResourceType` is `nil` for whatever ARM reason.  Therefore it was causing the panic since we were trying to convert the nil pointer to a string value.